src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1744,6 +1745,87 @@ class Simplex_tree {
     }
   }
 
+ public:
+  /** \brief Serialize the Simplex tree - Flatten it in a vector of char */
+  std::vector<char> serialize() {
+    std::vector<char> buffer{};
+    rec_serialize(&root_, buffer);
+    return buffer;
+  }
+
+ private:
+  /** \brief Serialize each element of the sibling and recursively call serialization. */
+  void rec_serialize(Siblings *sib, std::vector<char>& buffer) {
+    Gudhi::simplex_tree::serialize(static_cast<Vertex_handle>(sib->members().size()), buffer);
+#ifdef DEBUG_TRACES
+    std::clog << "\n" << sib->members().size() << " : ";
+#endif  // DEBUG_TRACES
+    for (auto& map_el : sib->members()) {
+      Gudhi::simplex_tree::serialize(map_el.first, buffer); // Vertex
+      Gudhi::simplex_tree::serialize(map_el.second.filtration(), buffer); // Filtration
+#ifdef DEBUG_TRACES
+      std::clog << " [ " << map_el.first << " | " << map_el.second.filtration() << " ] ";
+#endif  // DEBUG_TRACES
+    }
+    for (auto& map_el : sib->members()) {
+      if (has_children(&map_el)) {
+        rec_serialize(map_el.second.children(), buffer);
+      } else {
+        Gudhi::simplex_tree::serialize(static_cast<Vertex_handle>(0), buffer);
+#ifdef DEBUG_TRACES
+        std::cout << "\n0 : ";
+#endif  // DEBUG_TRACES
+      }
+    }
+  }
+
+ public:
+  /** \brief Deserialize the vector of char (flatten version of the tree) to create and return a Simplex tree */
+  static Simplex_tree* deserialize(std::vector<char>& buffer) {
+    Simplex_tree* stree = new Simplex_tree();
+    // Needs to read size before recursivity to manage new siblings for children
+    Vertex_handle members_size{};
+    std::vector<char>::const_iterator opositional_itr = Gudhi::simplex_tree::deserialize(std::cbegin(buffer), members_size);
+    opositional_itr = stree->rec_deserialize(&(stree->root_), members_size, opositional_itr, 0);
+    GUDHI_CHECK(opositional_itr == std::cend(buffer), std::out_of_range("Deserialization do not match end of buffer"));
+    return stree;
+  }
+
+ private:
+  /** \brief Serialize each element of the sibling and recursively call serialization. */
+  std::vector<char>::const_iterator rec_deserialize(Siblings *sib,
+                                                    Vertex_handle members_size,
+                                                    std::vector<char>::const_iterator ipositional_itr,
+                                                    int dim) {
+    std::vector<char>::const_iterator opositional_itr = ipositional_itr;
```

Comment:
Why does it suddenly change name?
I am not a big fan of the name (say I am oposed to it :stuck_out_tongue: ), an iterator always indicates a position.